install-deps.sh

```diff
@@ -339,7 +339,7 @@ else
         case "$ID" in
             fedora)
                 $SUDO dnf install -y dnf-utils
-                $SUDO dnf install -y docker-ce docker-ce-cli containerd.io
+                $SUDO dnf install -y moby-engine containerd
```

Comment:
Thanks @rishabh-d-dave . Am using Fedora 32 and am getting error while installing moby-engine ` package docker-ce-cli-1:19.03.12-3.fc31.x86_64 conflicts with docker provided by moby-engine-19.03.11-1.ce.git42e35e6.fc32.x86_64`. IMO we need moby-engine from fedora 33 onwards. So I think this should work better-
```suggestion
            if [ $VERSION_ID -lt 33 ] ; then
              $SUDO dnf install -y dnf-utils
              $SUDO dnf install -y docker-ce docker-ce-cli containerd.io
              $SUDO systemctl start docker
              $SUDO systemctl enable docker
            else
              $SUDO dnf install -y dnf-utils
              $SUDO dnf install -y moby-engine containerd
              $SUDO systemctl start docker
              $SUDO systemctl enable docker
            fi ;;
```

Reply:
@aaSharma14  Are you sure? I get same error on Fedora 32 VM -
```
$ sudo ./install-deps.sh 
Using dnf to install dependencies
Fedora Modular 32 - x86_64                      5.0 kB/s | 6.6 kB     00:01    
Fedora Modular 32 - x86_64 - Updates            5.6 kB/s | 5.0 kB     00:00    
Fedora Modular 32 - x86_64 - Updates             30 kB/s | 296 kB     00:10    
Fedora 32 - x86_64 - Updates                    6.6 kB/s | 4.9 kB     00:00    
Fedora 32 - x86_64 - Updates                    128 kB/s | 3.7 MB     00:29    
Last metadata expiration check: 0:00:01 ago on Sat Mar 13 11:34:59 2021.
Package dnf-utils-4.0.18-1.fc32.noarch is already installed.
Dependencies resolved.
Nothing to do.
Complete!
Last metadata expiration check: 0:00:21 ago on Sat Mar 13 11:34:59 2021.
No match for argument: docker-ce
No match for argument: docker-ce-cli
No match for argument: containerd.io
Error: Unable to find a match: docker-ce docker-ce-cli containerd.io
$ cat /etc/redhat-release 
Fedora release 32 (Thirty Two)
```

Reply:
> @aaSharma14 Are you sure? I get same error on Fedora 32 VM -
> 
> ```
> $ sudo ./install-deps.sh 
> Using dnf to install dependencies
> Fedora Modular 32 - x86_64                      5.0 kB/s | 6.6 kB     00:01    
> Fedora Modular 32 - x86_64 - Updates            5.6 kB/s | 5.0 kB     00:00    
> Fedora Modular 32 - x86_64 - Updates             30 kB/s | 296 kB     00:10    
> Fedora 32 - x86_64 - Updates                    6.6 kB/s | 4.9 kB     00:00    
> Fedora 32 - x86_64 - Updates                    128 kB/s | 3.7 MB     00:29    
> Last metadata expiration check: 0:00:01 ago on Sat Mar 13 11:34:59 2021.
> Package dnf-utils-4.0.18-1.fc32.noarch is already installed.
> Dependencies resolved.
> Nothing to do.
> Complete!
> Last metadata expiration check: 0:00:21 ago on Sat Mar 13 11:34:59 2021.
> No match for argument: docker-ce
> No match for argument: docker-ce-cli
> No match for argument: containerd.io
> Error: Unable to find a match: docker-ce docker-ce-cli containerd.io
> $ cat /etc/redhat-release 
> Fedora release 32 (Thirty Two)
> ```
> 
> Can you please recheck?

@aaSharma14 ping.

Reply:
I cant see any error with this on my Fedora32 - 
$SUDO dnf install -y dnf-utils
if rpm -q docker-ce || rpm -q docker-ce-cli || rpm -q moby-engine || rpm -q podman; then
echo "docker is already installed"
else
$SUDO dnf install -y moby-engine containerd
fi
$SUDO systemctl start docker
$SUDO systemctl enable docker

Reply:
@aaSharma14 Oh, OK. So you already have docker-ce repo installed on your system. I see why you don't get those errors in f32 now.

Reply:
Yes actually, because in this case docker provided by docker-ce conflicts with docker provided by moby-engine. So if we can just check if docker exists already then we should bypass the installation else proceed with moby-engine( which IMO works with all the fedora versions).

Reply:
I got that already. It would've been faster if you had told me why/how did you have docker-ce available on your Fedora 32 machine before. :)

Anyways, thanks!

Reply:
@aaSharma14 
Please check if the current patch works for you.

Interestingly, `install-deps.sh` right now installs docker packages only for Fedora and does nothing for RHEL, CentOS, etc. Is there a reason behind? It's more correct if we install docker for entire OS family if we are installing it for one of the family members.

Reply:
Thanks @rishabh-d-dave . It works fine now. I think I just missed that part adding docker to other distros. Adding the same patch just below - 

**centos|rhel|ol|virtuozzo)
MAJOR_VERSION="$(echo $VERSION_ID | cut -d. -f1)"
$SUDO dnf install -y dnf-utils**
dnf list installed | egrep "docker-ce|docker-ce-cli|containerd\.io"
if test $? -ne 0; then
$SUDO dnf install -y moby-engine containerd
 fi
$SUDO systemctl start docker
$SUDO systemctl enable docker

should do this right?